ui/src/app/kvm/views/KVMList/KVMListTable/KVMListTable.tsx

```diff
@@ -52,49 +139,147 @@ const KVMListTable = (): JSX.Element => {
     dispatch(zoneActions.fetch());
   }, [dispatch]);
 
+  // Update sort parameters depending on whether the same sort key was clicked.
+  const updateSort = (newSortKey: Sort["key"]) => {
```

Comment:
Do you think this function or any other bits of functionality could be shared via utils or hooks with other tables like the machine list?

Reply:
Definitely! I initially wanted it to be generic but in the machine list all the data needed for sorting exists on the machines themselves, which isn't the case for the KVM list. I'll try and figure out a good way to abstract out the sorting functions in a way that can work for both.

Reply:
Happy for that to be an update if you'd like to land all this work first. This branch is already rather large.

Reply:
Yeah that works for me. Maybe it would be a good idea to do that once selecting is done too we we can create hooks/utils for both.

Reply:
@Caleb-Ellis yep that sounds reasonable to me, maybe just make a card to remember.

Reply:
Opened an issue here: #1279